File: src/main/scala/com/crypteron/scala/showcase/model/Patient.scala
package com.crypteron.scala.showcase.model

import scala.annotation.meta._
import javax.persistence._

@Entity
class Patient(@(Id @field)@(GeneratedValue @field)(strategy = GenerationType.AUTO) var id: Int,
    var name: String = null,
    var secureSSN: String = null) {
  def this() = this(id = 0)
}

case class PatientDto(id: Int = 0, name: String, ssn: String)

File: src/main/java/com/crypteron/showcase/model/Patient.java
package com.crypteron.showcase.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import com.crypteron.Secure;

@Entity
public class Patient {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private int    id;
  @Secure
  private String firstName;
  @Secure
  private String lastName;
  @Secure
  private String ssn;

  public int getId() {
    return this.id;
  }

  public void setId(final int id) {
    this.id = id;
  }

  public String getFirstName() {
    return this.firstName;
  }

  public void setFirstName(final String firstName) {
    this.firstName = firstName;
  }

  public String getLastName() {
    return this.lastName;
  }

  public void setLastName(final String lastName) {
    this.lastName = lastName;
  }

  public String getSsn() {
    return this.ssn;
  }

  public void setSsn(final String ssn) {
    this.ssn = ssn;
  }

}
